### PR TITLE
Adding Snapshot Taking document to public

### PR DESCRIPTION
**Pull request by Phanco**
### How was it solved?

- Added `Taking_Snapshot.md`, with 2 extra mentions:
1. excluded_address not available yet
2.  instruction to adjust snapshot height
- README.md updated to reflect new file

### CODE REVIEW

File: documentation/Taking_Snapshot.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+## Generate Merkle Tree
+
+After running the following command, `accounts.json`, `merkle-root.json`, `merkle-tree-result.json` and `merkle-tree-result-detailed.json` will be generated.
+The descriptions of the above files can be found at [Tech Design](./Tech_Design.md) and [Tree Builder README](../packages/tree-builder/README.md).
+
+### Lisk Token Migration
+
+Generate merkle tree for Lisk Token Migration.
+
+<span style="color:red">🔴 excluded-addresses-path feature will be available when Snapshot Height has reached. i.e. 21 May, 2024 (0800 CET) 🔴</span>.
```
Comment (Phanco) [inline documentation/Taking_Snapshot.md]: seems that html tag not working in github but other MD reader

File: documentation/Taking_Snapshot.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+- jq
+
+## Preparation
+
+1. Synchronize blockchain using `lisk-core`. The node will automatically shut down itself after about a round after the Snapshot Height (#24,823,618) has been finalized, i.e. block #24,823,721 is finalized. Under ideal conditions, this might take about ~40-45 mins after the block at Snapshot Height is generated.
+   After synchronization, the snapshot data for the specified height (`system.backup.height`) in the Lisk Core node config can be found in `./snapshot/backup/`.
+
+   > **Synchronization from scratch may take up to days.**
+
+   > **To test this functionality before Snapshot Height has reached, you can edit `system.backup.height` to a smaller amount at file `config/mainnet/config.json`**
```
Comment (nagdahimanshu) [inline documentation/Taking_Snapshot.md]:
```suggestion
   > **To test this functionality before Snapshot Height has reached, you can edit `system.backup.height` to a lower height at file `config/mainnet/config.json`**
```

Comment (sameersubudhi) [inline documentation/Taking_Snapshot.md]:
```suggestion
   > **To test this functionality before the actual Snapshot Height, please update `system.backup.height` to a lower height in `config/mainnet/config.json` and restart the node with the `--overwrite-config` flag**
```

File: documentation/Taking_Snapshot.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+   git clone -b v4.0.6 https://github.com/LiskHQ/lisk-core.git && cd lisk-core
+
+   # Verify Backup height is correct and in place
+   cat config/mainnet/config.json | jq ".system.backup.height"   # 24823618
+
+   # Install dependencies and build lisk-core
+   yarn install --frozen-lockfile && yarn build
+
+   # Start lisk-core and let it keep running until synchronization has completed and reached block #24,823,618
+   # Estimated Available Date: 21 May, 2024 (0800 CET)
```
Comment (sameersubudhi) [inline documentation/Taking_Snapshot.md]:
```suggestion
   # Initial Estimated Date: 21 May, 2024 (0800 CEST)
```

File: documentation/Taking_Snapshot.md (hunk 1 above)
Comment (sameersubudhi) [inline documentation/Taking_Snapshot.md]:
```suggestion
```

File: documentation/Taking_Snapshot.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+| ----------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | -------- | ------------------ |
+| db-path                 | Database path, where your state.db is located. If following the preparation stage above, db-path would be `../../../`                                                          | True     |                    |
+| output-path             | Destination path of the merkle tree                                                                                                                                            | False    | `./data`           |
+| token-id                | Token ID, use default for mainnet LSK Token                                                                                                                                    | False    | `0000000000000000` |
+| excluded-addresses-path | File Path of List of addresses excluded from Migration. Exact addresses to be excluded from Migration has been stored in `lisk-token-claim/data/mainnet/exclude_addresses.txt` | False    | `""`               |
```
Comment (sameersubudhi) [inline documentation/Taking_Snapshot.md]:
```suggestion
| excluded-addresses-path | File Path of List of addresses excluded from Migration. Exact addresses to be excluded from Migration has been stored in `lisk-token-claim/data/mainnet/exclude_addresses.txt` <br> <span style="color:red">🔴 This flag will be made available in a subsequent release, expected after the Lisk L1 Mainnet Snapshot Height (#24,823,618) has been finalized 🔴</span> | False    | `""`               |
```

File: documentation/Taking_Snapshot.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+```
+# Create a separate folder to store Merkle Tree for Migration
+mkdir -p ./migration
+
+# excluded-addresses-path feature will be available when Snapshot Height has reached. i.e. 21 May, 2024 (0800 CET)
```
Comment (sameersubudhi) [inline documentation/Taking_Snapshot.md]:
```suggestion
# 🔴 `--excluded-addresses-path` flag will be made available in a subsequent release, expected after the Lisk L1 Mainnet Snapshot Height (#24,823,618) has been finalized 🔴
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

**Review by sameersubudhi — APPROVED** [review]
LGTM 👍

**Review by nagdahimanshu — APPROVED** [review]

**Pull request merged**